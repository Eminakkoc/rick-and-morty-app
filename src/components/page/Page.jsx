import React from 'react';
import PropTypes from 'prop-types';

import { LanguageConsumer } from 'common/language/LanguageContext';
import languages from 'common/language/languages';

const Page = (props) => {
  const { children, backTransition } = props;

  return (
    <LanguageConsumer>
      {({ language }) => (
        <div className="page-container">
          <div className="page-header">
            <div className="page-logo" />
          </div>
          {
            backTransition ? (
              <button
                type="button"
                className="page-back-button"
                onClick={backTransition}
              >
                {`< ${languages[language].back_button}`}
              </button>
            ) : null
          }
          {children}
        </div>
      )}
    </LanguageConsumer>
  );
};

Page.propTypes = {
  children: PropTypes.node,
  backTransition: PropTypes.func,
};

Page.defaultProps = {
  children: null,
  backTransition: null,
};

export default Page;
