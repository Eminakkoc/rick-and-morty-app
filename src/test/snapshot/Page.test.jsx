// CharacterCard.test.js
import React from 'react';
import Page from '../../components/page/Page';
import renderer from 'react-test-renderer';

test('Info Pair Component Structure', () => {
  const component = renderer.create(
    <Page />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
