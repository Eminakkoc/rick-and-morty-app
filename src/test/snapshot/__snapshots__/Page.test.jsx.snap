// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Pair Component Structure 1`] = `
<div
  className="page-container"
>
  <div
    className="page-header"
  >
    <div
      className="page-logo"
    />
  </div>
</div>
`;
